ompi/communicator/comm_cid.c

```diff
@@ -878,11 +898,14 @@ int ompi_comm_activate_nb (ompi_communicator_t **newcomm, ompi_communicator_t *c
         OMPI_COMM_SET_PML_ADDED(*newcomm);
     }
 
-    /* Step 1: the barrier, after which it is allowed to
-     * send messages over the new communicator
+    /**
+     * Dual-purpose barrier:
+     * 1. The communicator's disjointness is inferred fromimax_local_peers.
+     * 2. After the operation it is allowed to send messages over the new communicator.
+     *
      */
-    ret = context->allreduce_fn (&context->ok, &context->ok, 1, MPI_MIN, context,
```

Comment:
This call is illegal in MPI because one should explicitly use MPI_IN_PLACE as the send buffer if the send and recv buffer is the same for a collective. I do realize that this issue was there before, but we need to address it.

Reply:
Thanks for raising this - does `MPI_IN_PLACE` apply to the internal allreduce though? I'm fuzzy on the differences from `MPI_Allreduce`

Reply:
Absolutely. There are no differences with `MPI_Allreduce`, same API, same requirements, but in some cases we can use a non-MPI function (such as a PRRTE centralized id generator). I think we did not see this issue because the reduction happens on a single int, and this will in all cases take the eager pt2pt path. Regardless, the code is incorrect as is right now.

Reply:
Please see update. I realized that this allreduce can also happen on inter-communicators, so we cannot use MPI_IN_PLACE. Instead I used a temp variable for sendbuf.